test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -80,12 +88,63 @@ spec = Test.group "Meta-Value Manipulation" <|
         Meta.is_an err Error . should_be_true
         Meta.is_a err Text . should_be_false
 
+        Meta.is_a Date.now Date.Date . should_be_true
+        Meta.is_a Date_Time.now Date_Time.Date_Time . should_be_true
+        Meta.is_a Date_Time.now Date.Date . should_be_false
+        Meta.is_a Time_Of_Day.now Time_Of_Day.Time_Of_Day . should_be_true
+        Meta.is_a Time_Of_Day.now Date.Date . should_be_false
+        Meta.is_a Date_Time.now.zone Time_Zone.Time_Zone . should_be_true
+        Meta.is_a Date_Time.now.zone Date.Date . should_be_false
+
+
+    Test.specify "should allow for returning the type of a value" <|
+        Meta.type_of 42 . should_equal Integer
+        Meta.type_of 2.81 . should_equal Decimal
+        Meta.type_of 2.81 . should_not_equal Integer
+
+        # Can't compare like
+        # Meta.type_of [1,2,3] . should_equal Vector.Vector
+        # because equality check tries to compare lengths, where RHS is a type
+        v_tpe = Meta.type_of [1,2,3]
+        v_tpe.is_of_type Vector.Vector . should_be_true
+        v_tpe.is_of_type Array . should_be_false
+        [1,2,3].to_array.is_of_type Array . should_be_true
+
+        Meta.type_of "foo" . should_equal Text
+        Meta.type_of "0" . should_not_equal Integer
+
+        Meta.type_of True . should_equal Boolean
+        Meta.type_of False . should_not_equal Any
+
+        (Meta.type_of Date.now).is_of_type Date.Date . should_be_true
+        (Meta.type_of Date.now).is_of_type Date_Time.Date_Time . should_be_false
+        (Meta.type_of Date_Time.now).is_of_type Date_Time.Date_Time . should_be_true
+        (Meta.type_of Date_Time.now).is_of_type Date.Date . should_be_false
+        (Meta.type_of Time_Of_Day.now).is_of_type Time_Of_Day.Time_Of_Day . should_be_true
+        (Meta.type_of Time_Of_Day.now).is_of_type Date.Date . should_be_false
+        (Meta.type_of Date_Time.now.zone).is_of_type Time_Zone.Time_Zone . should_be_true
+        (Meta.type_of Date_Time.now.zone).is_of_type Date.Date . should_be_false
+
+        list = ArrayList.new
+        list.add 123
+        list_tpe = Meta.type_of list
+        list_tpe . should_not_equal JObject
+        list_tpe . should_equal ArrayList
+        list_tpe.is_of_type ArrayList . should_be_true
+
+        e = IOException.new "meh"
+        e_tpe = Meta.type_of e
+        e_tpe . should_equal IOException
+        e_tpe . should_not_equal JException
+        e_tpe.is_of_type IOException . should_be_true
+        e_tpe.is_of_type JException . should_be_false
```

Comment:
Maybe worth adding `is_a` checks too?

Reply:
Ok, I noticed that this is planned for a separate PR. Sorry then, ignore this comment.